xgi/classes/simplicialcomplex.py

```diff
@@ -214,32 +218,52 @@ def add_simplex(self, members, id=None, **attr):
         if not members:
             raise XGIError("Cannot add an empty edge")
 
-        if not self.has_simplex(members):
+        if self.has_simplex(members):
+            return
 
-            if id in self._edge.keys():  # check that uid is not present yet
-                warn(f"uid {id} already exists, cannot add simplex {members}")
-                return
-
-            uid = next(self._edge_uid) if not id else id
-            self._edge[uid] = set()
-            for node in members:
-                if node not in self._node:
-                    if node is None:
-                        raise ValueError("None cannot be a node")
-                    self._node[node] = set()
-                    self._node_attr[node] = self._node_attr_dict_factory()
-                self._node[node].add(uid)
-
-            self._edge[uid] = members
-            self._edge_attr[uid] = self._hyperedge_attr_dict_factory()
-            self._edge_attr[uid].update(attr)
-
-            if id:  # set self._edge_uid correctly
-                update_uid_counter(self, id)
+        if id in self._edge.keys():  # check that uid is not present yet
+            warn(f"uid {id} already exists, cannot add simplex {members}")
+            return
+
+        uid = next(self._edge_uid) if not id else id
+        self._edge[uid] = set()
+        for node in members:
+            if node not in self._node:
+                if node is None:
+                    raise ValueError("None cannot be a node")
+                self._node[node] = set()
+                self._node_attr[node] = self._node_attr_dict_factory()
+            self._node[node].add(uid)
+
+        self._edge[uid] = members
+        self._edge_attr[uid] = self._hyperedge_attr_dict_factory()
+        self._edge_attr[uid].update(attr)
+
+        if id:  # set self._edge_uid correctly
+            update_uid_counter(self, id)
+
+        # add all subfaces
+        faces = self._subfaces(members)
+        faces = set(faces)  # get unique faces
+        for members in faces:
```

Comment:
Is it a problem that `members` from the input is getting overwritten here?

Reply:
Probably not but I changed it to be sure, thanks